rclcpp/include/rclcpp/subscription_options.hpp

```diff
@@ -77,6 +77,10 @@ struct SubscriptionOptionsBase
     // Topic statistics publication period in ms. Defaults to one second.
     // Only values greater than zero are allowed.
     std::chrono::milliseconds publish_period{std::chrono::seconds(1)};
+
+    // An optional QoS which can provide topic_statistics with a stable QoS separate from
+    // the subscription's current QoS settings which could be unstable.
+    rclcpp::QoS statistics_qos_configuration{10};
```

Comment:
a couple of things need to be added with this.

- use this QoS setting to statistics publisher.

https://github.com/ros2/rclcpp/blob/7c1143dc1502d5dda99a3dfa17124a72dbaea90c/rclcpp/include/rclcpp/create_subscription.hpp#L92

- How about using `SystemDefaultsQoS` (or some other QoS profile) instead of using hard coded number?

https://github.com/ros2/rclcpp/blob/7c1143dc1502d5dda99a3dfa17124a72dbaea90c/rclcpp/include/rclcpp/qos.hpp#L482-L502